### PR TITLE
docs(js/react-router): Add setup guide for react router (framework)

### PR DESCRIPTION
**Pull request by chargome**
This PR adds react-router (v7 framework mode) as a new guide within the javascript platform.

Since we do not yet provide a dedicated SDK, these docs just cover a very minimal setup using `@sentry-react` and `@sentry/node`.

closes https://github.com/getsentry/sentry-javascript/issues/14578

### CODE REVIEW

File: platform-includes/getting-started-config/javascript.react-router.mdx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import { hydrateRoot } from "react-dom/client";
+import { HydratedRouter } from "react-router/dom";
+
+Sentry.init({
+  dsn: "___PUBLIC_DSN___",
+  integrations: [Sentry.browserTracingIntegration()],
+
+  tracesSampleRate: 1.0, //  Capture 100% of the transactions
+
+  // Set `tracePropagationTargets` to control for which URLs trace propagation should be enabled
```
Comment (coolguyzone) [inline platform-includes/getting-started-config/javascript.react-router.mdx:25]:
```suggestion
  // Set `tracePropagationTargets` to declare which URL(s) should have trace propagation enabled
```

File: platform-includes/getting-started-config/javascript.react-router.mdx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  hydrateRoot(
+    document,
+    <StrictMode>
+      <HydratedRouter />
+    </StrictMode>
+  );
+});
+```
+
+Now, update your `app/root.tsx` file to include a report any unhandled errors from your error boundary:
```
Comment (coolguyzone) [inline platform-includes/getting-started-config/javascript.react-router.mdx:39]:
```suggestion
Now, update your `app/root.tsx` file to include a report on any unhandled errors from your error boundary:
```

File: platform-includes/getting-started-config/javascript.react-router.mdx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+### Server-Side Setup
+
+Create a `instrument.server.mjs` file in the root of your app:
+
+```js {filename: instrument.server.mjs}
+import * as Sentry from "@sentry/node";
+
+Sentry.init({
+  dsn: "https://c122e7637b24dc299bbef9d4e80c8325@o4507458071298048.ingest.de.sentry.io/4508415844876368",
```
Comment (coolguyzone) [inline platform-includes/getting-started-config/javascript.react-router.mdx:87]: Looks like you have a hardcoded DSN here, is this correct?

File: platform-includes/getting-started-sourcemaps/javascript.react-router.mdx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+## Add Readable Stack Traces to Errors
+
+By default, React Router will minify your JavaScript and CSS files in production. This makes it difficult to debug errors in production. To make debugging easier, you can generate source maps and upload them to Sentry.
+
+We recommend to use [Sentry's Vite plugin](/platforms/javascript/sourcemaps/uploading/vite/) to upload sourcemaps.
+
+Please refer to the <PlatformLink to="/sourcemaps">Sourcemaps Documentation</PlatformLink>, for more information.
```
Comment (coolguyzone) [inline platform-includes/getting-started-sourcemaps/javascript.react-router.mdx:7]:
```suggestion
Please refer to the <PlatformLink to="/sourcemaps">Source Maps Documentation</PlatformLink>, for more information.
```

In most of our docs we use 2 words for 'source maps', let's stick to that convention 🤙

File: platform-includes/getting-started-sourcemaps/javascript.react-router.mdx [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+For more advanced configuration, you can use [`sentry-cli`](https://github.com/getsentry/sentry-cli) directly to upload sourcemaps.
```
Comment (coolguyzone) [inline platform-includes/getting-started-sourcemaps/javascript.react-router.mdx:9]:
```suggestion
For more advanced configuration, you can use [`sentry-cli`](https://github.com/getsentry/sentry-cli) directly to upload source maps.
```

File: src/components/platformIcon.tsx [hunk 6]
```diff
@@ -273,6 +273,8 @@ import VueSVGLarge from 'platformicons/svg_80x80/vue.svg';
 import WasmSVGLarge from 'platformicons/svg_80x80/wasm.svg';
 import WindowsSVGLarge from 'platformicons/svg_80x80/windows.svg';
 import XamarinSVGLarge from 'platformicons/svg_80x80/xamarin.svg';
+// TODO(cg): move to platformicons once available
```
Comment (coolguyzone) [inline src/components/platformIcon.tsx:276]: Just double-checking, you meant to commit this TODO right?

**Review by coolguyzone — APPROVED** [review]
Thanks for this big update @chargome! 🏄‍♂️ Looks good from a docs perspective, this is a pretty technical one so we should probably have an engineer review it as well.

File: platform-includes/getting-started-verify/javascript.react-router.mdx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+Throw an error in a loader to verify that Sentry is working.
+After opening this route in your browser, you should see two errors in the Sentry issue stream, one captured from the server and one captured from the client.
+
+```tsx {filename: error.tsx}
+import type { Route } from "./+types/error-loader";
+
+export async function loader() {
+  console.log("server loader");
+  await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (andreiborza) [inline platform-includes/getting-started-verify/javascript.react-router.mdx:9]:
```suggestion
```
l: I'd simplify this snippet, no reason to really wait here right?

**Review by andreiborza — APPROVED** [review]

File: src/components/platformIcon.tsx [hunk 8]
```diff
@@ -273,6 +273,8 @@ import VueSVGLarge from 'platformicons/svg_80x80/vue.svg';
...
+import ReactRouterSVGLarge from 'public/icon/rr.svg';
```
Comment (AbhiPrasad) [inline src/components/platformIcon.tsx:277]: could we just update platformicons first before merging in this PR? I'd rather avoid this temporary code, and I think these docs can wait the extra 2-3 days.

File: platform-includes/getting-started-config/javascript.react-router.mdx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+```tsx {filename: entry.client.tsx} {"onboardingOptions": {"performance": "9, 12-16", "session-replay": "10, 17-21"}}
+import * as Sentry from "@sentry/react";
+import { startTransition, StrictMode } from "react";
+import { hydrateRoot } from "react-dom/client";
+import { HydratedRouter } from "react-router/dom";
+
+Sentry.init({
+  dsn: "___PUBLIC_DSN___",
+  integrations: [
+    Sentry.browserTracingIntegration(),
```
Comment (AbhiPrasad) [inline platform-includes/getting-started-config/javascript.react-router.mdx:22]: shouldn't this use the react router integration?

File: src/components/platformIcon.tsx (hunk 8 above)
Comment (chargome) [reply src/components/platformIcon.tsx:277]: sure, I'll ping them again

File: platform-includes/getting-started-config/javascript.react-router.mdx (hunk 9 above)
Comment (chargome) [reply platform-includes/getting-started-config/javascript.react-router.mdx:22]: No as the routing mechanism is different when using framework mode. If you rely on the routes module then these functions that we need for our "library mode integration" are not exposed

**Review by andreiborza — APPROVED** [review]

**Pull request merged**